headerbp/metrics.go

```diff
@@ -17,6 +17,15 @@ const (
 )
 
 var (
+	clientMiddlewareIdempotencyCheckTotal = promauto.With(prometheusbpint.GlobalRegistry).NewCounterVec(prometheus.CounterOpts{
+		Name: "baseplate_client_middleware_idempotency_check_total",
```

Comment:
this is a metrics inside package `headerbp`, but the name is a bit too broad for what it does. I'd probably rename it into (`checks` also should be in plural form):
```suggestion
		Name: "baseplate_headerbp_client_middleware_idempotency_checks_total",
```
but also, this is only emitted when check fails, not when you checks (you don't emit it when check passes), so the "unit" should not be "checks" but something like "failures":
```suggestion
		Name: "baseplate_headerbp_client_middleware_idempotency_check_failures_total",
```
re @mathyourlife-reddit: none of the other middlewares will cause harm when added twice (prometheus metrics can cause double counter when added twice, but that's the limit of what "harm" they can cause, and it's possible to add prometheus metric middleware multiple times with different names, for example with retries), so I don't foresee we'll add idempotency check to other middlewares.

Reply:
no other current middleware cause issues if added twice (except for prometheus), but others in the future might.  Having a more general metric with the label gives us some flexibility in alerting/graphs instead of needing to duplicate in the future.

Reply:
I'm gonna keep it more generic but add `_failures`

Reply:
🔕 with a generic name we probably should define this metric in a common package then.

but we don't currently have a common package appropriate for this, so I guess leave it here for now is fine, we can move it to a different package later if we ever need to use it for something not related to headerbp.